u2-routing-controllers/features/tempController.js

```diff
@@ -0,0 +1,21 @@
+
+const getTemp = (req, res) => {
+  const u1 = req.params.u1.toUpperCase()
+  const u2 = req.params.u2.toUpperCase()
+  const t1 = req.params.t1
+
+  let t2
+  if (u1 === u2) {
+    t2 = t1
+  } else if (u1 === 'C') {
+    t2 = t1 * 9 / 5 + 32
+  } else if (u1 === 'F') {
+    t2 = (t1 - 32) * 5 / 9
+  }
+
+  res.send(`${parseFloat(t1).toFixed(1)} ${u1} = ${parseFloat(t2).toFixed(1)} ${u2}`)
```

Comment:
จะดีกว่าที่จะแยกวิเคราะห์ตัวเลขทันที นี่เป็นความผิดของฉัน ฉันคิดว่าฉันไม่ได้ทดสอบวิธีแก้ปัญหาอย่างถูกต้อง ฉันแนะนำสิ่งนี้ / It's better to parse the number immediately. This is my fault -- I think I didn't test my solution properly. I recommend:
```js
const t1 = parseFloat(req.params.t1)
...
res.send(`${t1.toFixed(1)} ${u1} = ${t2.toFixed(1)} ${u2}`)
```

`t2` เป็น float (floating-point number) โดยอัตโนมัติเนื่องจากวิธีคำนวณ
`t2` is automatically a float (floating-point number) due to how it's calculated